src/scikit_build_cli/__main__.py

```diff
@@ -1,14 +1,83 @@
 from __future__ import annotations
 
-from . import skbuild
+import pathlib
+from collections.abc import MutableMapping, Sequence
+from importlib.metadata import EntryPoint
 
+import click
 
-def run_cli() -> None:
```

Comment:
I think there is a design with making `run_cli` a thin wrapper around click command
https://github.com/teemtee/tmt/blob/7d67df001cf52c464e5706d80ad1cd09f583c66f/tmt/__main__.py#L12
I think it is to better catch import errors, loading of the entry_points. Didn't fully understand how it is meant to be designed

Reply:
Since this is a standalone package, it always has its dependencies (unless it's a broken install, but then we can rely on normal Python errors, that's not our fault). All commands are loaded when displaying help, but only the command required is loaded if you just run one command, potentially making it faster.

This is one of the three advertised points for click: https://click.palletsprojects.com/en/8.1.x/  :)

Reply:
Unless there's a reason to, I don't think we should have a public API. A plugin doesn't need to import anything, just add an entry point, so it's not needed for that.

Reply:
What about errors coming from external cli-plugin?

Reply:
About public api, it would allow to load multiple sub-commands by pointing the entry-point to the module instead of the commands themselves. (By setting `@skbuild.command` instead of `@click.commamd`). It should work with this lazy-loader as well.

Reply:
If you load modules, you have to load the module to see what commands it offers. While if you don't, you only need the entry-points to see what commands are available and select the one you want. I don't see much benefit for allowing a plugin to provide one entry point and several sub-commands when it can just statically list the subcommands as entry points and we get a more responsive interface and simpler implementation. The only downside is you have to rerun `pip install -e.` when adding a command, but that only affects us. :)

We could also allow `skbuild <command>` to call `skbuild-<command>`, like git & cargo.

(I also don't know if we really expect many or any plugins?)

PS: I also like avoiding too many wrappers or custom commands. The more it looks like vanilla click (or whatever), the easier it is for people familiar with that framework but not our package to work with it. I'm not against adding any wrappers, but best IMO to keep them focused and only when necessary.

Reply:
FYI, if you look at my implementation of repo-review, it's designed around entry-points, you have one per set of checks (it loads a dict), one per fixture, and one for families. Seems a little overboard, but it keeps the implementation clean, and being able to easily add support to a project without adding a dependency allowed it to be used in a minimal-dependency project (validate-pyproject)!.